### PR TITLE
docs: README for shares package

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/718

### CODE REVIEW

File: pkg/shares/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Shares
+
+Package shares provides primitives for splitting block data into shares and
+parsing shares back into block data.
```
Comment (SweeXordious) [inline pkg/shares/README.md:4]: [Non Blocking][Suggestion]
I guess that editors nowadays handle soft line wrapping perfectly. Thus, no need to add new lines in the middle of a sentence.

File: pkg/shares/README.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+motivation behind the distinction is that transactions, intermediate state
+roots, and evidence are expected to have small lengths so they are encoded in
+compact shares to minimize the number of shares needed to store them. On the
+other hand, messages are expected to be larger and have the desideratum that
+clients should be able to create proofs of message inclusion. This desiradum is
+infeasible if client A's message is encoded into a share with another client B's
+message that is unknown to A. It follows that client A's message is encoded into
+a share such that the contents can be determined by client A without any
+additional information. See [message layout
+rational](https://celestiaorg.github.io/celestia-specs/latest/rationale/message_block_layout.html#message-layout-rationale)
```
Comment (SweeXordious) [inline pkg/shares/README.md:25]: [Non Blocking][Suggestion]
Probably link also to the ADR as it is easier to understand.

**Review by SweeXordious — APPROVED** [review]
LGTM. Only non blocking suggestions

**Review by evan-forbes — APPROVED** [review]
very well said! this is a huge value add imo

**Pull request merged**